architectures/firmware/sof-zephyr/app_layer/index.rst

```diff
@@ -3,38 +3,53 @@
 Application Layer
 #################
 
-Application Layer represents the built-in FW processing components, loadable FW
-components and example templates with application libraries required for
-components integration. Application layer content is assumed to be open source
+Application Layer represents the built-in FW modules (processing components), loadable FW
+modules and example templates with application libraries required for
+modules integration. Application layer content is assumed to be open source
 and only proprietary 3rd party components should remain private.
 
 .. uml:: images/app_layer_diagram.pu
    :caption: Application Layer
 
-Components in Application Layer
-*******************************
+Modules in Application Layer
+****************************
 
-The built-in components are built together with base firmware and they have
+The built-in modules are built together with base firmware and they have
 direct access to all firmware drivers and service APIs. When built-in module
 is enabled in configuration, it is guaranteed to exist in firmware binary.
 
-The loadable components are built separately from base firmware and they are
+The loadable modules are built separately from base firmware and they are
 loaded dynamically as a separate binary, depending on the host audio
-configuration.
+configuration. For building is used LMDK(Loadable Module Development Kit)
+which is standalone kit containing all required files.
 
-All the application layer components access base firmware services via System
+All the application layer module access base firmware services via System
 Services ABI.
 
-**NOTE:** The built-in components are utility components provided by base
+**NOTE:** The built-in modules are utility components provided by base
 firmware/kernel.
 
-Examples of built-in components:
+Examples of built-in modules:
 
 * Audio built-in components: Copiers, Mixers, Volume, SRC, etc.
 
+Example how to build loadable module:
+
+* Example Up-Down-Mixer build using :ref:`lmdk_user_guide`
+
 Probe
 =====
 
 The probe module is special module in FW infrastructure that allows to inject
 or extract data from a specified probe point. The traditional client
 platforms use HDA DMAs to transfer data in and out of such module.
+
+Loadable Modules
+================
+
+The loadable modules are build to separate binaries. To communicate with them
+is used native system service and to control is used module api. Data flow is based on
+IADK modules and uses module adapter as a transision between module api and processing component api.
```

Comment:
please link or define IADK

uses the module adapter

"transision" is a typo if you mean `transition` then I am not sure what this sentence means